lib/helpers.js

```diff
@@ -210,15 +210,21 @@ async function getApkanalyzerForOs (sysHelpers) {
  * in comparison to `adb dumpsys window` output parsing.
  * But the trust command does not work for `Swipe` unlock pattern.
  *
- * In some Android devices (Probably around Android 10+) show `mCurrentFocus` as `NotificationShade` or `StatusBar`
- * as the lock screen while they do not have mShowingLockscreen nor mDreamingLockscreen.
- * Although, these `NotificationShade` and `StatusBar` could be also regular notification view.
+ * In some Android devices (Probably around Android 10+), `mShowingLockscreen` and `mDreamingLockscreen`
+ * do not work to detect lock status. Instead, keyguard preferences helps to detect the lock condition.
+ * Some devices such as Android TV do not have keyguard, so we should keep
+ * screen condition as this primary method.
  *
  * @param {string} dumpsys - The output of dumpsys window command.
  * @return {boolean} True if lock screen is showing.
  */
 function isShowingLockscreen (dumpsys) {
-  return /(mShowingLockscreen=true|mDreamingLockscreen=true|mCurrentFocus=.+NotificationShade|mCurrentFocus=.+StatusBar)/gi.test(dumpsys);
+  if (/(mShowingLockscreen=true|mDreamingLockscreen=true)/gi.test(dumpsys)) {
+    return true;
+  }
+  // Fallback to KeyguardServiceDelegate preference.
+  // mIsShowing and mInputRestricted are true in lock condition.
+  return dumpsys.includes('mIsShowing=false', 'mInputRestricted=false');
```

Comment:
return _.every(['mIsShowing=false', 'mInputRestricted=false'], (x) -> dumpsys.includes(x))

Reply:
Oh, it's cool, `some` and `every` methods